### PR TITLE
[MM-29106, MM-34847, MM-34843, MM-34846] Collapsed reply threads, Global Threads & In Channel experience UI implementation

### PR DESCRIPTION
**Pull request by ashishbhate**
#### Summary
Implements Collapsed Reply Threads UI

#### Ticket Link
[MM-29106](https://mattermost.atlassian.net/browse/MM-29106), [MM-34847](https://mattermost.atlassian.net/browse/MM-34847), [MM-34843](https://mattermost.atlassian.net/browse/MM-34843), [MM-34846](https://mattermost.atlassian.net/browse/MM-34846)

#### Checklist
- [x] Global Threads 
- [x] In Channel Experience
- [x] Includes text changes and localization file updates
- [ ] New messages line in threads 

#### Device Information
This PR was tested on: iPhone 12 (iOS 14.5, Simulator) & Pixel 2 (Android 11, Emulator) 

#### Screenshots
Collage

#### Release Note
```release-note
Added support for collapsed reply threads on mobile.
```

### CODE REVIEW

**Comment by ashishbhate** [comment]
This PR is not even close to being reviewable. I'll ping when its ready for review.

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Really great work @anurag6713 and @ashishbhate. There are a number of UI tweaks that we need to make which I will post in a separate message, but let me identify some of the more glaring things I found first.

1. If you have no unreads in the global threads view, we need to visually disable the 'mark all as unread' icon button. Right now it still looks tappable, but doesn't do anything. 
2. If I am viewing a direct message thread, the header displays differently and doesn't have the option to Follow/Unfollow in the header. I understand why you've done that, but it presents an inconsistency with the web app. 
3. At one point, I tapped on the 'Threads' menu item from the sidebar and got 'No followed threads yet' screen, but I obviously have hundreds of threads that I follow. I don't seem to have clear steps to reproduce unfortunately, but I've definitely seen it a couple times now.
4. I don't know if I can describe the steps to reproduce this, but I was browsing my threads view and then somehow just got a white screen. Then I closed and re-opened the app and got a bit error message (see screenshot). Not sure if this is somehow connected to #3 above.

5. If I am on Global Threads and I switch teams, it always shows the Threads view. I would expect it to display the last visited page from that team instead (like we do on the web app)
6. The thread view shows 'UX Design' no matter what channel the thread belongs to. Seems like it's hard-coded.

The UI tweaks are in this [follow-up ticket](https://mattermost.atlassian.net/browse/MM-36852)

**Comment by jgilliam17** [comment]
Thanks @anurag6713 
I'm seeing the same crash as Matt. 
For me, it happened when switching teams while on global threads after scroll bar got switched to left. I don't have steps on how to repro the scrollbar issue, it is not reproducing consistently.
IMG_9242

**Comment by matthewbirtch** [comment]
@anurag6713 I created a follow-up ticket for the minor UI polish items: https://mattermost.atlassian.net/browse/MM-36852

**Comment by enahum** [comment]
Enter to see the threads, there are X amount showed in the hamburger menu & the sidebar item, go to unreads, mark them all as read. unreads indicator is showing and hamburger menu keeps the same number you had before.
image

Also seeing the same issue as described here https://community.mattermost.com/core/pl/uz9wb8j1qjdc7eiyttk6q8h41o

Let me know if you want to take another look before I start the review process.

**Comment by enahum** [comment]
Threads view being displayed on the device, go to webapp and disable.

two issues: React trying to update state on an unmounted component and the bigger one is a crash
image

**Comment by enahum** [comment]
removing myself as a reviewer from the time being until this issues are addresssed.

Btw Miguel is out until August, please try and not assign any reviews to him.

**Comment by esethna** [comment]
@enahum please help prioritize your review here as you can, thanks!

**Comment by jgilliam17** [comment]
Thanks @anurag6713 
Switching teams crash is resolved for me.
Seeing few issues
1. "No followed threads" screen was presented at least once, despite having multiple threads
2. Font theme is off for global threads screen when switching teams while viewing global threads on both teams
IMG_A13D46C2D5B0-1

**Comment by esethna** [comment]
@enahum if you can help with a review today that would be much appreciated!

**Review by koox00 — APPROVED** [review]
Nice work!

**Review by matthewbirtch — APPROVED** [review]

**Comment by matthewbirtch** [comment]
I'm approving this knowing that we have a few other follow-up tickets to resolve some of the other UI tweaks/improvements

**Comment by enahum** [comment]
> @enahum if you can help with a review today that would be much appreciated!

@esethna you are asking me to review this PR, but this PR does not tell the entire story of the feature. is there another PR that I need to review first?

**Comment by esethna** [comment]
@enahum Ah, here's the other: https://github.com/mattermost/mattermost-mobile/pull/5175

**Review by esethna — APPROVED** [review]
I've been using the PR build for a few days, works great!

**Comment by jgilliam17** [comment]
@anurag6713 Is this ticket [MM-36852](https://mattermost.atlassian.net/browse/MM-36852) also addressed here or in a separate PR? Thanks.

**Comment by anurag6713** [comment]
> @anurag6713 Is this ticket [MM-36852](https://mattermost.atlassian.net/browse/MM-36852) also addressed here or in a separate PR? Thanks.

Yes @jgilliam17

**Comment by esethna** [comment]
@enahum @jgilliam17 please help with review here :)

**Comment by jgilliam17** [comment]
Hello @anurag6713 (tested using test app found [here](https://community-daily.mattermost.com/core/pl/d4pwrht99tbq5m5tse7z1m85te))
I am seeing few issue with following a thread, thread you have not participated/followed before
1. After following a thread, my avatar was added to root post footer

https://user-images.githubusercontent.com/52937121/126003980-5830b9ac-000c-493f-8310-ebbdc3873996.mov

2. When following a single post without any replies, some random avatars are added to the footer

https://user-images.githubusercontent.com/52937121/126004147-3a67606d-c6ca-4f33-9d94-d6febafc8f10.MOV

File: app/screens/post_options/post_options.js [hunk 1]
```diff
@@ -285,6 +309,12 @@ export default class PostOptions extends PureComponent {
         });
     };
 
+    handleToggleFollow = () => {
+        const {actions, currentTeamId, currentUserId, post} = this.props;
+        actions.setThreadFollow(currentUserId, currentTeamId, post.id, !post.is_following);
```
Comment (enahum) [inline app/screens/post_options/post_options.js:314]: careful with the currentTeamId, this should be the team that the post belongs to and only the currentTeamId if the post is in a DM/GM.

I guess you can actually fix an existing bug here with the `copy link` option ;)

File: app/screens/thread/index.js [hunk 2]
```diff
@@ -4,29 +4,36 @@
...
 
-import {getTheme} from '@mm-redux/selectors/entities/preferences';
-
 import {selectPost} from '@mm-redux/actions/posts';
+import {setThreadFollow, updateThreadRead} from '@mm-redux/actions/threads';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
 import {getChannel, getMyCurrentChannelMembership} from '@mm-redux/selectors/entities/channels';
 import {makeGetPostIdsForThread} from '@mm-redux/selectors/entities/posts';
+import {getTheme, isCollapsedThreadsEnabled} from '@mm-redux/selectors/entities/preferences';
+import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
```
Comment (enahum) [inline app/screens/thread/index.js:13]: same issue about the currentTeamId as the others I described

File: app/screens/thread/index.js [hunk 3]
```diff
@@ -4,29 +4,36 @@
...
             channelId: ownProps.channelId,
+            channelIsArchived: channel ? channel.delete_at !== 0 : false,
             channelType: channel ? channel.type : '',
+            collapsedThreadsEnabled,
             displayName: channel ? channel.display_name : '',
             myMember: getMyCurrentChannelMembership(state),
-            rootId: ownProps.rootId,
             postIds: getPostIdsForThread(state, ownProps.rootId),
+            rootId: ownProps.rootId,
+            teamId: getCurrentTeamId(state),
```
Comment (enahum) [inline app/screens/thread/index.js:32]:
```suggestion
            teamId: channel?.team_id || getCurrentTeamId(state),
```

File: app/screens/thread/index.js [hunk 4]
```diff
@@ -4,29 +4,36 @@
...
         return {
             channelId: ownProps.channelId,
+            channelIsArchived: channel ? channel.delete_at !== 0 : false,
             channelType: channel ? channel.type : '',
+            collapsedThreadsEnabled,
             displayName: channel ? channel.display_name : '',
             myMember: getMyCurrentChannelMembership(state),
-            rootId: ownProps.rootId,
             postIds: getPostIdsForThread(state, ownProps.rootId),
+            rootId: ownProps.rootId,
```
Comment (enahum) [inline app/screens/thread/index.js:31]: no need for this, it is already being passed to the component

File: app/screens/thread/index.js [hunk 5]
```diff
@@ -4,29 +4,36 @@
...
+            teamId: getCurrentTeamId(state),
             theme: getTheme(state),
-            channelIsArchived: channel ? channel.delete_at !== 0 : false,
+            thread: getThread(state, ownProps.rootId, true),
             threadLoadingStatus: state.requests.posts.getPostThread,
+            userId: getCurrentUserId(state),
```
Comment (enahum) [inline app/screens/thread/index.js:36]: perhaps make this clearer as in 
```suggestion
            currentUserId: getCurrentUserId(state),
```

File: app/screens/thread/thread_base.js [hunk 6]
```diff
@@ -14,21 +15,29 @@ import DeletedPost from 'app/components/deleted_post';
 import {popTopScreen, mergeNavigationOptions} from 'app/actions/navigation';
 import {TYPING_HEIGHT, TYPING_VISIBLE} from '@constants/post_draft';
 
+import ThreadFollow from './thread_follow';
+
+Navigation.registerComponent('ThreadFollow', () => ThreadFollow);
```
Comment (enahum) [inline app/screens/thread/thread_base.js:20]: why is this being registered here and not in screens/index ? I'd like to keep the registration of Navigation components in one place

File: app/screens/thread/thread_base.js [hunk 7]
```diff
@@ -14,21 +15,29 @@ import DeletedPost from 'app/components/deleted_post';
...
         channelType: PropTypes.string,
+        collapsedThreadsEnabled: PropTypes.bool,
         displayName: PropTypes.string,
         myMember: PropTypes.object.isRequired,
+        postIds: PropTypes.array.isRequired,
         rootId: PropTypes.string.isRequired,
+        teamId: PropTypes.string.isRequired,
         theme: PropTypes.object.isRequired,
-        postIds: PropTypes.array.isRequired,
-        channelIsArchived: PropTypes.bool,
```
Comment (enahum) [inline app/screens/thread/thread_base.js:30]: why was this removed?

File: app/screens/thread/thread_follow/index.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+type Props = {
+    active: boolean;
+    intl: typeof IntlShape;
+    onPress: () => void;
+    theme: Theme
+};
+
+function ThreadFollow({active, intl, onPress, theme}: Props) {
+    const styles = getStyleSheet(theme);
+    const [containerStyle] = React.useMemo(() => {
```
Comment (enahum) [inline app/screens/thread/thread_follow/index.tsx:22]: what's the benefit of memoizing here?

**Review by enahum — CHANGES_REQUESTED** [review]
I see that there is no change in the `getPostIdsInCurrentChannel` for the channel post list, and this is fine for now as you are forcing a `wipe` of the entire redux store, but do please take this in consideration for when we fix the `wipe` issue

File: app/components/avatars/avatars.tsx [hunk 9]
```diff
@@ -101,15 +102,16 @@ export default class Avatars extends PureComponent<AvatarsProps> {
...
-        const {userIds, breakAt, theme} = this.props;
+        const {userIds, breakAt, style: baseContainerStyle, theme} = this.props;
         const displayUserIds = userIds.slice(0, breakAt);
         const overflowUsersCount = Math.min(userIds.length - displayUserIds.length, OVERFLOW_DISPLAY_LIMIT);
 
         const style = getStyleSheet(theme);
 
         return (
-            <TouchableWithFeedback
+            <TouchableOpacity
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/avatars/avatars.tsx:112]: To have the same experience on press like rest of the elements with opacity. As avatars on posts change opacity on press, changed it to make the experience similar.

[the remark this answers is not in the document]
Comment (enahum) [reply app/components/avatars/avatars.tsx:112]: Was this experience approved by UX? this is using `opacity` for iOS but the bubble effect on Android, are we changing Android? why?

[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/avatars/avatars.tsx:112]: My mistake .. I don't think we are changing for android. Will revert .. Thank you.

File: app/components/global_threads/global_threads.tsx [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+    const [isLoading, setIsLoading] = React.useState<boolean>(true);
+
+    const scrollToTop = React.useCallback(() => {
+        listRef.current?.scrollToOffset({offset: 0});
+    }, []);
+
+    const loadThreads = React.useCallback(async (before = '', after = '', unread = false) => {
+        setIsLoading(true);
+        await actions.getThreads(userId, teamId, before, after, undefined, false, unread);
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/global_threads.tsx:46]: We are going to replace all the threads in the store with the latest on first request thus eliminating the state/deleted threads.

On App init or going to background, we are only maintaining last 60 threads.

In future implementation, we will sync with `deleted` as `true` and remove the deleted threads.

File: app/components/global_threads/thread_item/thread_item.tsx [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+                    style={style.unreadDot}
+                    testID={`${testIDPrefix}.unread_dot`}
+                />
+            );
+        }
+    }
+
+    let name;
+    if (postItem.state === Posts.POST_DELETED) {
+        name = (
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/thread_item/thread_item.tsx:109]: When root message is deleted, users will still be able to navigate to the thread and read other messages.

File: app/components/global_threads/thread_item/thread_item.tsx [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+    }
+
+    let postBody;
+    if (postItem.state !== Posts.POST_DELETED) {
+        postBody = (
+            <Text
+                style={style.message}
+                numberOfLines={2}
+            >
+                <RemoveMarkdown value={postItem.message || ''}/>
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/thread_item/thread_item.tsx:134]: We are limiting the number of lines to 2 and rendering markdown might make the height as variable which should be then controlled with `overflow` `hidden`.

File: app/components/sidebars/main/threads_entry/threads_entry.tsx [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+    const theme = useSelector((state: GlobalState) => getTheme(state));
+    const currentUserId = useSelector((state: GlobalState) => getCurrentUserId(state));
+    const currentTeamId = useSelector((state: GlobalState) => getCurrentTeamId(state));
+
+    const threadCount = useSelector((state:GlobalState) => getTeamThreadCounts(state, currentTeamId));
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        dispatch(getThreads(currentUserId, currentTeamId, '', '', undefined, false, isUnreadSelected));
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/sidebars/main/threads_entry/threads_entry.tsx:44]: Code has changed. In global screen mount, we are loading 5 threads (to get count of total threads, unreads, mentions).

On global threads, we are loading more threads.

File: app/components/sidebars/main/threads_entry/threads_entry.tsx [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+
+    const onPress = preventDoubleTap(() => {
+        EventEmitter.emit(NavigationTypes.CLOSE_MAIN_SIDEBAR);
+        dispatch(handleViewingGlobalThreadsScreen());
+    });
+
+    const style = getStyleSheet(theme);
+    const extraStyle = getExtraStyleSheet(theme);
+
+    const [itemStyle, textStyle, iconStyle, border, badge] = React.useMemo(() => {
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/sidebars/main/threads_entry/threads_entry.tsx:55]: To prevent sending the new object `{}` to the components when there is no change.

File: app/screens/channel/channel.ios.js [hunk 15]
```diff
@@ -53,17 +54,27 @@ export default class ChannelIOS extends ChannelBase {
...
-            renderDraftArea = true;
+        if (viewingGlobalThreads) {
             component = (
-                <>
-                    <ChannelPostList registerTypingAnimation={this.registerTypingAnimation}/>
-                </>
+                <GlobalThreadsList/>
             );
+        } else {
+            safeAreaEdges.push('bottom');
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/screens/channel/channel.ios.js:68]: UI/UX requires we need to show this as we are showing our sidebar.

File: app/components/global_threads/thread_item/thread_item.tsx [hunk 16]
```diff
@@ -0,0 +1,273 @@
...
+    const threadStarterName = displayUsername(threadStarter, Preferences.DISPLAY_PREFER_FULL_NAME);
+
+    const showThread = () => {
+        actions.getPostThread(postItem.id);
+        actions.selectPost(postItem.id);
+        const passProps = {
+            channelId: postItem.channel_id,
+            rootId: postItem.id,
+        };
+        goToScreen(THREAD, '', passProps);
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/thread_item/thread_item.tsx:75]: This isn't working as the `EventEmitter.on` is not available at the time this code is running in global threads screen.

File: app/components/global_threads/thread_footer/thread_footer.tsx [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+import {UserProfile} from '@mm-redux/types/users';
+import {$ID} from '@mm-redux/types/utilities';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+const NO_PARTICIPANTS: object[] = [];
+
+export type DispatchProps = {
+    actions: {
+        setThreadFollow: (currentUserId: $ID<UserProfile>, currentTeamId: $ID<Team>, threadId: $ID<UserThread>, newState: boolean) => void;
```
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/thread_footer/thread_footer.tsx]: Thanks Elias! Getting the teamId in action for all the possible thread actions now 🙂

File: app/components/avatars/avatars.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/avatars/avatars.tsx]: from UI/UX, I was asked to remove the dark colour appearing on press. I have added `type={'opacity'}` to remove that on iOS & android.

File: app/components/global_threads/global_threads.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/global_threads/global_threads.tsx:50]: sounds good, I guess this is part of the same ticket to avoid wiping the store, yes?

[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/global_threads.tsx:50]: Yes Elias .. it covers parts related to sync and removing stale threads as we are depending on `wiping` for now.

File: app/components/global_threads/thread_footer/index.ts [hunk 18]
```diff
@@ -0,0 +1,30 @@
...
+import {setThreadFollow} from '@mm-redux/actions/threads';
+import {getCurrentUserId} from '@mm-redux/selectors/entities/common';
+import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
+import type {GlobalState} from '@mm-redux/types/store';
+
+import ThreadFooter, {StateProps, DispatchProps, OwnProps} from './thread_footer';
+
+function mapStateToProps(state: GlobalState) {
+    return {
+        currentTeamId: getCurrentTeamId(state),
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/global_threads/thread_footer/index.ts:16]: Resolving as now this component seems not to mount anywhere other than the global threads and the channel screen

[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/thread_footer/index.ts]: This one is fixed in the latest push. We are getting team id in action instead of sending from component as you have suggested. I think it makes sense and also keeps the code clean. I might try doing the same for other things. Thanks.

**Comment by esethna** [comment]
@enahum @anurag6713 I think you guys have this covered but if you can please help rally reviews so we get CRT onto master for v1.46, then cut a beta

cc// @amyblais

File: app/components/global_threads/thread_item/thread_item.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/global_threads/thread_item/thread_item.tsx:129]: But then this means that I would see a message like 
```
### this is a header
`some text` and a [url](https://someurl.com)
```

@matthewbirtch is that expected?

File: app/components/sidebars/main/channels_list/channel_item/index.js [hunk 19]
```diff
@@ -80,12 +84,13 @@ function makeMapStateToProps() {
             isChannelMuted: isChannelMuted(member),
             isGuest,
             isManualUnread: isManuallyUnread(state, ownProps.channelId),
-            mentions: member ? member.mention_count : 0,
+            mentions: member?.mention_count_root || 0,
```
Comment (enahum) [inline app/components/sidebars/main/channels_list/channel_item/index.js:87]: is this always mention_count_root even when collapse threads in not enabled?

File: app/components/sidebars/main/channels_list/list/list.js [hunk 20]
```diff
@@ -42,6 +43,7 @@ export default class List extends PureComponent {
         canJoinPublicChannels: PropTypes.bool.isRequired,
         canCreatePrivateChannels: PropTypes.bool.isRequired,
         canCreatePublicChannels: PropTypes.bool.isRequired,
+        collapsedThreadsEnabled: PropTypes.bool.isRequired,
```
Comment (enahum) [inline app/components/sidebars/main/channels_list/list/list.js:46]: where is this prop being used?

File: app/components/sidebars/main/threads_entry/index.tsx [hunk 21]
```diff
@@ -0,0 +1,144 @@
...
+import {changeOpacity} from '@utils/theme';
+import {makeStyleFromTheme} from '@mm-redux/utils/theme_utils';
+import Badge from '@components/badge';
+
+type Props = {
+    intl: typeof intlShape;
+};
+
+const ThreadsEntry = ({intl}: Props) => {
+    const viewingGlobalThreads = useSelector(getViewingGlobalThreads);
```
Comment (enahum) [inline app/components/sidebars/main/threads_entry/index.tsx:33]: no need to do it now, but I want to see this changed in a later PR where we get rid of `useSelector` this will cause massive pain when porting to Gekidou

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/components/global_threads/thread_item/thread_item.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/thread_item/thread_item.tsx:104]: https://mattermost.atlassian.net/browse/MM-35887

File: app/components/global_threads/thread_item/thread_item.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (anurag6713) [reply app/components/global_threads/thread_item/thread_item.tsx:129]: It will appear like this .. Reusing the same component which removes markdown like Pinned messages in channel header.
Screenshot 2021-07-20 at 13 42 09

File: app/screens/thread/index.js (hunk 2 above)
Comment (anurag6713) [reply app/screens/thread/index.js:13]: Moved getting teamId to action. Will remove this line if it is unused in the actions.

File: app/components/sidebars/main/channels_list/channel_item/index.js (hunk 19 above)
Comment (anurag6713) [reply app/components/sidebars/main/channels_list/channel_item/index.js:87]: I don't think so. Good catch .. thank you 🙂

File: app/components/sidebars/main/threads_entry/index.tsx (hunk 21 above)
Comment (anurag6713) [reply app/components/sidebars/main/threads_entry/index.tsx:33]: Missed this one while refactoring global thread components. Fixed it now. Thank you.

File: app/screens/thread/thread.ios.test.js [hunk 22]
```diff
@@ -12,10 +12,23 @@ import {shallowWithIntl} from 'test/intl-test-helper';
...
+        Navigation: {
+            events: () => ({
+                registerAppLaunchedListener: jest.fn(),
+            }),
+            registerComponent: jest.fn(),
+            setLazyComponentRegistrator: jest.fn(),
+        },
+    };
+});
+
```
Comment (enahum) [inline app/screens/thread/thread.ios.test.js:26]: can you move this to https://github.com/mattermost/mattermost-mobile/blob/master/test/setup.js#L213

**Review by enahum — CHANGES_REQUESTED** [review]
one small change and it should be good to go

**Review by enahum — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @anurag6713 
Tested, looks good to merge.
- Verified CRT feature on the mobile.
Added few issues for follow up:
https://mattermost.atlassian.net/browse/MM-37294
https://mattermost.atlassian.net/browse/MM-37295
https://mattermost.atlassian.net/browse/MM-37296
https://mattermost.atlassian.net/browse/MM-37297
